OpenRA.Game/TextNotification.cs

```diff
@@ -0,0 +1,56 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2021 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using OpenRA.Primitives;
+
+namespace OpenRA
+{
+	public enum TextNotificationPool { System, Chat, Mission, Feedback }
+
+	public class TextNotification : IEquatable<TextNotification>
+	{
+		public readonly Color PrefixColor;
+		public readonly string Prefix;
+		public readonly string Text;
+		public readonly Color TextColor;
```

Comment:
Pedantic nit (feel free to ignore): we list (color, text, text, color) here compared to (text, text, color, color) in the ctor. Could we reorder these fields to either (text, color, text, color) or (text, text, color, color) for consistency?